src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key1 in obj.extraData) {
+          state[key1] = obj.extraData[key1];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key2 of obj.keysToRemove) {
```

Comment:
```suggestion
        for (const key of obj.keysToRemove) {
```

Reply:
it's other key :) Not you used before :)
For loop when you declare `const` it's creating variable for it own scope